File: app/src/main/java/com/example/qiuchenly/comicparse/MVP/Presenter/MainPresenter.java
package com.example.qiuchenly.comicparse.MVP.Presenter;

import com.example.qiuchenly.comicparse.Bean.HotComicStrut;
import com.example.qiuchenly.comicparse.MVP.Contract.MainContract;
import com.example.qiuchenly.comicparse.MVP.Model.Fragment_MainModel;
import com.example.qiuchenly.comicparse.Simple.BasePresenterImp;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

/**
 * 主页P层实现类
 * 作者 秋城落叶
 * 2018年7月30号
 */
public class MainPresenter extends BasePresenterImp<MainContract.View, Fragment_MainModel> implements MainContract.Presenter {

    public MainPresenter(MainContract.View mView) {
        super(mView);
    }

    @Override
    public void getHotComic() {
        SuperModel.getHotsComic(new MainContract.GetHotComic() {
            @Override
            public void onSuccessGetHot(@NotNull ArrayList<HotComicStrut> arr) {
                if (isShow()) SuperView.getHotComicList(arr);
            }

            @Override
            public void onFailed(@NotNull String reasonStr) {
                if (isShow()) SuperView.ShowErrorMsg(reasonStr);
            }
        });
    }

    @Override
    public Fragment_MainModel createModel() {
        return new Fragment_MainModel();
    }
}

File: ComicX/src/main/java/com/example/qiuchenly/comicparse/MVP/Presenter/MainPresenter.kt
package com.example.qiuchenly.comicparse.MVP.Presenter

import com.example.qiuchenly.comicparse.Bean.HotComicStrut
import com.example.qiuchenly.comicparse.MVP.Contract.MainContract
import com.example.qiuchenly.comicparse.MVP.Model.Fragment_MainModel
import com.example.qiuchenly.comicparse.Simple.BasePresenterImp

import java.util.ArrayList

/**
 * 主页P层实现类
 * 作者 秋城落叶
 * 2018年7月30号
 */
class MainPresenter(mView: MainContract.View) : BasePresenterImp<MainContract.View, Fragment_MainModel>(mView), MainContract.Presenter {

    override fun getHotComic() {
        SuperModel!!.getHotsComic(object : MainContract.GetHotComic {
            override fun onSuccessGetHot(arr: ArrayList<HotComicStrut>) {
                if (isShow) SuperView!!.getHotComicList(arr)
            }

            override fun onFailed(reasonStr: String) {
                if (isShow) SuperView!!.ShowErrorMsg(reasonStr)
            }
        })
    }

    override fun createModel(): Fragment_MainModel {
        return Fragment_MainModel()
    }
}